### PR TITLE
:bug: Fixes fromUri path + adds param during login and logout for navigation

### LINKED ISSUE #84: Document how to override the referrer to redirect to different page than login
Inspired by https://devforum.okta.com/t/implicit-callback-redirects-to-the-angular-component-and-then-to-homepage-automatically/960/6.

### PR DESCRIPTION
**Pull request by jmelberg-okta**
### Updates
- Automatically set `fromUri` value when the `AuthGuard` is triggered.
     - In the default case where no `onAuthRequired` is passed in, we pass the current `url` into the `loginRedirect` method. 
    - When `onAuthRequired` is passed in, the current url value must be set before executing the `onAuthRequired` method.
        - For example: `/protected` is blocked by the `OktaAuthGuard`, where a custom `onAuthRequired` logic navigates the user to `/login`.
- Update the default `OktaLoginRedirectComponent` to use either the default path `/` as the `fromUri` or the value provided by the `onAuthRequired` guard logic.
- Removes checking for the object type of scope (breaking change)
- Adds configurable url to redirect to after logout

### Resolves
#84 (Angular)

### CODE REVIEW

File: packages/okta-angular/README.md [hunk 1]
```diff
@@ -144,6 +171,52 @@ const oktaConfig = {
 };
 ```
 
+## Reference
+### `oktaAuth`
+`oktaAuth` is the top-most component of `okta-angular`. This is where most of the configuration is provided.
```
Comment (robertjd) [inline packages/okta-angular/README.md]: I'm finding this section a bit confusing because `oktaAuth` is a property that we add to the scope, after you've called `OktaAuthModule.initAuth(oktaConfig)` with the configuration options that your mentioning.  This isn't really clear, currently it looks like oktaAuth is a method I should call with the configuration options?

Comment (jmelberg-okta) [reply packages/okta-angular/README.md]: +1. This is a bit confusing.

File: packages/okta-angular/README.md [hunk 2]
```diff
@@ -144,6 +171,52 @@ const oktaConfig = {
...
+If you have an Okta `sessionToken`, you can bypass the full-page redirect by passing in this token. This is recommended when using the [Okta Sign-In Widget](https://github.com/okta/okta-signin-widget). Simply pass in a `sessionToken` into the `loginRedirect` method follows:
+
+```typescript
+// Navigate to "/profile" on login success
+this.oktaAuth.loginRedirect('/profile', {
+  sessionToken: /* sessionToken */
+})
+```
+
+> Note: For information on obtaining a `sessionToken` using the [Okta Sign-In Widget](https://github.com/okta/okta-signin-widget), please see the [`renderEl()` example](https://github.com/okta/okta-signin-widget#rendereloptions-success-error).
```
Comment (robertjd) [inline packages/okta-angular/README.md]: I know we're trying to wrap Auth JS and eventually take away the direct access, but we should put some kind of documentation in here about that these "additional params" are - even if it's just linking to the AuthJS method we're delegating to: https://github.com/okta/okta-auth-js#tokengetwithredirectoptions

File: packages/okta-angular/src/okta/okta.service.ts [hunk 3]
```diff
@@ -147,21 +152,21 @@ export class OktaAuthService {
...
-    async logout() {
+    async logout(uri?: String) {
       this.oktaAuth.tokenManager.clear();
       await this.oktaAuth.signOut();
+      this.router.navigate([uri || '/']);
     }
 
     /**
      * Scrub scopes to ensure 'openid' is included
+     * @param scopes
```
Comment (robertjd) [inline packages/okta-angular/src/okta/okta.service.ts]: These JSDoc comments need more fidelity, for example:

```
@param {String} uri The location to navigate to after logout
```

File: packages/okta-angular/src/okta/okta.service.ts [hunk 4]
```diff
@@ -97,8 +97,13 @@ export class OktaAuthService {
 
     /**
      * Launches the login redirect.
+     * @param fromUri
+     * @param additionalParams
      */
-    loginRedirect(additionalParams?: object) {
+    loginRedirect(fromUri: String, additionalParams?: object) {
```
Comment (robertjd) [inline packages/okta-angular/src/okta/okta.service.ts]: This signature change is also a breaking change

Comment (jmelberg-okta) [reply packages/okta-angular/src/okta/okta.service.ts]: Should this be pulled out as another PR?

File: packages/okta-angular/src/okta/okta.service.ts (hunk 3 above)
Comment (robertjd) [reply packages/okta-angular/src/okta/okta.service.ts]: Or even better cuz this is typescript, just update the method:

```
scrubScope(scopes: string)
```

File: packages/okta-angular/README.md [hunk 5]
```diff
@@ -144,6 +171,86 @@ const oktaConfig = {
...
+#### `oktaAuth.getAccessToken()`
+Returns the access token from storage (if it exists).
+
+#### `oktaAuth.getIdToken()`
+Returns the ID token from storage (if it exists).
+
+#### `oktaAuth.getUser()`
+Returns the result of the OpenID Connect `/userinfo` endpoint if an access token exists.
+
+#### `oktaAuth.handleAuthentication()`
```
Comment (robertjd) [inline packages/okta-angular/README.md]: Also: this does not return and redirects you to the stored URL

File: packages/okta-angular/README.md [hunk 6]
```diff
@@ -144,6 +171,86 @@ const oktaConfig = {
...
+Parses the tokens returned as hash fragments in the OAuth 2.0 Redirect URI.
+
+### `oktaAuth.logout(uri?)`
```
Comment (robertjd) [inline packages/okta-angular/README.md]: Missing a 4th # on this header

File: packages/okta-angular/README.md [hunk 7]
```diff
@@ -144,6 +171,86 @@ const oktaConfig = {
 };
 ```
 
+## Reference
```
Comment (robertjd) [inline packages/okta-angular/README.md]: I think this reference section needs another pass.  Right now it appears to be serving multiple purposes:

- Show you all the available classes you can import
- Tells you about some configuration options (but doesn't tell you where they go or how to use them)
- Shows you how to do configure the final redirect URL for login and logout operations
- Documents some, but not all, of the public methods available on the `OktaAuthModule` service

The content in here is good, though I feel that it needs to be moved around to other sections, and that this "reference" section should be more focused on describing the `OktaAuthModule` and what it provides you.

File: packages/okta-angular/src/okta/okta.guard.ts [hunk 8]
```diff
@@ -46,20 +46,17 @@ export class OktaAuthGuard implements CanActivate {
...
 
-    /** 
-     * Store the current path
-     */
-    this.oktaAuth.setFromUri(state.url);
-
     /**
      * Redirect to the given path or
      * perform the default Okta full-page redirect.
      */
```
Comment (robertjd) [inline packages/okta-angular/src/okta/okta.guard.ts]: Does this comment block still make sense?

File: packages/okta-angular/src/okta/okta.service.ts [hunk 9]
```diff
@@ -147,21 +152,21 @@ export class OktaAuthService {
...
      * Scrub scopes to ensure 'openid' is included
+     * @param scopes
      */
-    scrubScopes(scopes) {
+    scrubScopes(scopes: string) {
       if (!scopes) {
         return 'openid email';
-      } else {
-        // Make sure object is a string
-        scopes = Array.isArray(scopes) ? scopes.join(' ') : scopes
```
Comment (robertjd) [inline packages/okta-angular/src/okta/okta.service.ts]: Before merging, let's continue our conversation about conventional commits.  I think this breaking change needs to be it's own commit, and the change to loginRedirect() needs to be it's own commit.  There's a lot going on in this PR and someone scanning for breaking changes may have a hard time finding them.

**Comment by jmelberg-okta** [comment]
Closing due to #125
